source/core/parameters_lagrangian.cc

```diff
@@ -437,6 +478,53 @@ namespace Parameters
         list_y = Utilities::string_to_double(y_str_list);
         list_z = Utilities::string_to_double(z_str_list);
 
+        // Read vx, vv and vz list as a single string
+        std::string vx_str = prm.get("list velocity x");
+        std::string vy_str = prm.get("list velocity y");
+        std::string vz_str = prm.get("list velocity z");
+
+
+        // Convert vx,vy andv vz string to vector of strings
+        std::vector<std::string> vx_str_list(
+          Utilities::split_string_list(vx_str));
+        std::vector<std::string> vy_str_list(
+          Utilities::split_string_list(vy_str));
+        std::vector<std::string> vz_str_list(
+          Utilities::split_string_list(vz_str));
+
+        // Convert vx,vy and vz string vector to double vector
+        list_vx = Utilities::string_to_double(vx_str_list);
+        list_vy = Utilities::string_to_double(vy_str_list);
+        list_vz = Utilities::string_to_double(vz_str_list);
+
+        // Read wx, wy and wz list as a single string
+        std::string wx_str = prm.get("list omega x");
+        std::string wy_str = prm.get("list omega y");
+        std::string wz_str = prm.get("list omega z");
+
+        // Convert wx,wy and wz string to vector of strings
+        std::vector<std::string> wx_str_list(
+          Utilities::split_string_list(wx_str));
+        std::vector<std::string> wy_str_list(
+          Utilities::split_string_list(wy_str));
+        std::vector<std::string> wz_str_list(
+          Utilities::split_string_list(wz_str));
+
+        // Convert x,y and z string vector to double vector
+        list_wx = Utilities::string_to_double(wx_str_list);
+        list_wy = Utilities::string_to_double(wy_str_list);
+        list_wz = Utilities::string_to_double(wz_str_list);
+
+        // Read the diameters list as a single string
+        std::string d_str = prm.get("list diameters");
+
+        // Convert the diameters list to a vector of strings
+        std::vector<std::string> d_str_list(
+          Utilities::split_string_list(d_str));
+
+        // Convert the diameters string vector to a double vector
+        list_d = Utilities::string_to_double(d_str_list);
```

Comment:
I think a good approach to list insertion would be to get the number of particles from the maximum size among vectors and then repeat the default value or a given single value for the remaining particles if values are not provided. Otherwise, you will always need the ``set list ... = 0.0, 0.0, 0.0......`` for each vector, and it is not necessary if you want to have the same value for all particles.